100Change2017/etl/compose-and-upload

```diff
@@ -0,0 +1,139 @@
+                                  will be uploaded to the torque wiki.  It needs to have subdirectories
+                                  by proposal number.
+
+  --tdc-config-dir DIR            DIR is the location for files that are the base configuration files
+                                  needed by TorqueDataConnect, and can be optionally, manually, put on
+                                  the torque wiki.  We don't autmoatically do that because we want to
+                                  overwrite the configuration out there.
+"""
+
+from etl import competition, wiki, toc, tdc
+import config
+import getopt
+import sys
+import os
+
+
+def main():
+    """Compose the MacFound input and emit it as html-ized csv."""
+    try:
+        opts, args = getopt.getopt(
+            sys.argv[1:],
+            "",
+            [
+                "proposals-csv=",
+                "tdc-config-dir=",
+                "attachments-dir=",
+            ],
+        )
+    except getopt.GetoptError as err:
+        sys.stderr.write("ERROR: '%s'\n" % err)
+        sys.exit(2)
+
+    proposals_csv = None
+    attachments_dir = None
+    tdc_config_dir = None
+    for o, a in opts:
+        if o == "--proposals-csv":
+            proposals_csv = a
+        elif o == "--tdc-config-dir":
+            tdc_config_dir = a
+        elif o == "--attachments-dir":
+            attachments_dir = a
+        else:
+            sys.stderr.write("ERROR: unrecognized option '%s'\n" % o)
+            sys.exit(2)
+
+    if proposals_csv is None:
+        sys.stderr.write("ERROR: need --proposals-csv option.\n\n")
+        sys.stderr.write(__doc__)
+        sys.exit(1)
+
+    comp = competition.Competition(proposals_csv, "LFC100Change2017", "Review_Number")
+    comp.add_supplemental_information(
+        competition.MediaWikiTitleAdder("Registered Organization Name")
+    )
+
+    fix_cell_processor = competition.FixCellProcessor()
+    comp.process_all_cells_special(fix_cell_processor)
+    fix_cell_processor.report()
+
+    attachments = competition.RegexSpecifiedAttachments(
+        comp.sorted_proposal_keys, attachments_dir
+    )
+    attachments.specify_new_column("^\\d*_mou", "MOU Attachment")
+    attachments.specify_new_column("^\\d*_team", "Team Attachment")
+    attachments.specify_new_column("financial", "Financials Attachment")
+    comp.add_supplemental_information(attachments)
+
+    comp.sort("Registered Organization Name")
+
+    comp.add_toc(toc.GenericToc("Topic_TOC", "Primary Thematic Area"))
+
+    comp.process_tocs()
+
+    if tdc_config_dir is not None:
+        tdc.AllProposals(comp).generate(tdc_config_dir)
+        tdc.AllColumns(comp).generate(tdc_config_dir)
+        tdc.ProcessedSpreadsheet(comp).generate(tdc_config_dir)
+
+    my_wiki = wiki.WikiSession(
+        config.username, config.password, comp.name, config.wiki_url
+    )
+    my_wiki.upload_sheet(comp)
+    if os.environ.get("SKIP_ATTACHMENTS") is None:
```

Comment:
That's a good point.  It's mainly a way for me to do `SKIP_ATTACHMENTS=1 ./deploy ~/data` and not have it re upload all the attachments.  Like all things in this repository, it started as a way to do something in a bash script for LFC2020, and as things evolved, it got codified weirdly in python (why is there a random environment variable here?).  Realistically, it should be an argument passed into the top level deploy script that's passed along, as well as the pare option.

I think I'll make that change instead of just renaming it here.